### PR TITLE
Case insensitive SqliteDataReader. Fixes dotnet/efcore#24011

### LINKED ISSUE #24011: SqliteDataReader[<string>] is Case-Sensitive
The SqliteDataReader lookup when using a string is case-sensitive:
https://docs.microsoft.com/en-us/dotnet/api/microsoft.data.sqlite.sqlitedatareader.item?view=msdata-sqlite-5.0.0#Microsoft_Data_Sqlite_SqliteDataReader_Item_System_String_

Why? This isn't true of any of the other data readers that I've used including the Microsoft SQL Data reader.
https://docs.microsoft.com/en-us/dotnet/api/system.data.sqlclient.sqldatareader.item?view=dotnet-plat-ext-5.0#System_Data_SqlClient_SqlDataReader_Item_System_String_

To make matters worse, it is case-sensitive to the name of the column in the database, not what you select.

For example, if you have a table in your Sqlite database like the following:

`CREATE TABLE test (testId INT)`

The following code will error:

```C#
using (SqliteCommand command = new SqliteCommand("SELECT testID FROM test", conn))
{
  using (SqliteDataReader reader = command.ExecuteReader();
  {
    while (reader.read())
    {
      int testId = Convert.ToInt32(reader["testID"]);
    }
  } 
}
```

The lookup needs to use the same case as the table definition, not the query.
Microsoft.Data.Sqlite version: 5.0.2
Target framework: .Net Standard 2.0

### PR DESCRIPTION
**Pull request by wdesgardin**
Fixes #24011

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/efcore?pullRequest=24588) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/efcore?pullRequest=24588) before we can accept your contribution.<br/><br/>:x: wdesgardin [sign now](https://cla.dotnetfoundation.org/dotnet/efcore?pullRequest=24588)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/efcore?pullRequest=24588) it.</sub>

File: test/Microsoft.Data.Sqlite.Tests/SqliteDataReaderTest.cs [hunk 1]
```diff
@@ -1583,19 +1583,22 @@ public void Item_by_ordinal_throws_when_non_query()
             }
         }
 
-        [Fact]
-        public void Item_by_name_works()
+        [Theory]
+        [InlineData("SELECT 1 as Id;", "Id", 1L)]
+        [InlineData("SELECT 1 as Id;", "id", 1L)]
+        [InlineData("SELECT 1 as Id, 2 as id;", "id", 2L)]
```
Comment (bricelam) [inline test/Microsoft.Data.Sqlite.Tests/SqliteDataReaderTest.cs:1589]:
```csharp
[InlineData("SELECT 1 as Id, 2 as id;", "ID", 1L)]
```
...or should it throw?

**Review by bricelam — CHANGES_REQUESTED** [review]

File: test/Microsoft.Data.Sqlite.Tests/SqliteDataReaderTest.cs (hunk 1 above)
Comment (wdesgardin) [reply test/Microsoft.Data.Sqlite.Tests/SqliteDataReaderTest.cs:1589]: Hi @bricelam, thanks for reviewing this pr. Can you confirm that I need to open an issue to know if the team wants an exception to be raised in this particular case ?

Comment (bricelam) [reply test/Microsoft.Data.Sqlite.Tests/SqliteDataReaderTest.cs:1589]: We discussed this as a team today, and we think it should throw. Would you like to update PR? If not, I'm happy to merge this as is and make the update myself.

**Review by bricelam — APPROVED** [review]

**Pull request merged**